### DISCUSSION TITLE
Naming of select functions

### DISCUSSION THREAD

**Discussion by mtbeek32** [ObjectVision/GeoDMS#179] [General]
Within the set of selection functions, the following dimensions are relevant:
- function make a new unit and or relates attributes to this new unit
- function results in a org_rel attribute and uses this org_rel attribute to relate attributes to this new unit
- function results in an explicit value type

In the current set of functions, the selected options for these dimensions are sometimes not explicit.  
Therefore the following names are suggested:	

**old > new**
select_unit > select
select_unit_<I>valuetype</I> > select_<I>valuetype</I>

select_orgrel > select_org_rel
select_orgrel_<I>valuetype</I>> select_org_rel_<I>valuetype</I>

select_data > collect_data

select_many > select_relate
select_many_<I>valuetype</I> > select_relate_<I>valuetype</I>

select_afew> select_relate_org_rel
select_afew_<I>valuetype</I>> select_relate_org_rel_<I>valuetype</I>

**Comment by MaartenHilferink** [comment]
select[_vt] | select[_vt]_with_org_rel := depreciated select_unit | select_orgrel 
select[_vt]_with_attr_by_cond | select[_vt]_with_org_rel_with_attr_by_cond | select[_vt]_with_attr_by_org_rel := select_many | x | select_afew

collect_by_cond | collect_by_org_rel   :=    select_data | lookup
recollect_by_cond | recollect_by_org_rel

collect_attr_by_cond | collect_attr_by_org_rel := (depreciated) relate_many = relate_attr | relate_afew
recollect_attr_by_cond | recollect_attr_by_org_rel
